mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -104,46 +106,66 @@ void AffixTuningParameters::runOnFunction() {
   func.walk([&](miopen::GridwiseGemmOp op) {
     OpBuilder b(op.getContext());
 
-    // TBD. Compute tuning parameters from actual logic.
-    op.setAttr("block_size", b.getI32IntegerAttr(256));
+    ConvolutionContext convContext = populateConvContext(op);
+    InitParamsNonXDL validParams{0, 0, 0, 0, 0, 0};
+    GemmSize gemmSize;
+    DerivedParams gemmADerivedParam;
+    DerivedParams gemmBDerivedParam;
+    int64_t gemmCDstPerWrite;
+    int64_t gridSize;
+
+    PopulateParams populateParams;
+    LogicalResult status = populateParams.paramsFromCtx(
+        convContext, validParams, gemmSize, gemmADerivedParam,
+        gemmBDerivedParam, gemmCDstPerWrite, gridSize);
+    if (failed(status)) {
+      signalPassFailure();
+    }
 
-    op.setAttr("m_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("n_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("k_per_block", b.getI32IntegerAttr(8));
+    // TODO: Override the block size and grid size before populating params.
+    // The way it is implemented now cannot guarantee cohereancy of tuning
+    // parameters
+    int64_t grid_size = gridSize;
+    int64_t block_size = validParams.blockSize;
+    if (launchDimCallback) {
+      launchDimCallback(block_size, grid_size);
+    }
 
-    op.setAttr("m_per_thread", b.getI32IntegerAttr(4));
-    op.setAttr("n_per_thread", b.getI32IntegerAttr(4));
+    // Tunable parameters.
+    op.setAttr("block_size", b.getI32IntegerAttr(block_size));
```

Comment:
`block_size` -> `validParams.blockSize`?

Reply:
Refer to the reasoning above, will not fix it here. Will fix in the immediate next PR al-together.